SimGeneral/CaloAnalysis/plugins/CaloTruthAccumulator.cc

```diff
@@ -485,6 +485,8 @@ void CaloTruthAccumulator::finalizeEvent(edm::Event &event, edm::EventSetup cons
   } else {
     for (auto &sc : *(output_.pSimClusters)) {
       auto hitsAndEnergies = sc.hits_and_fractions();
+      for (auto &hAndE : hitsAndEnergies)
+        sc.addRecHitAndEnergy(hAndE.first, hAndE.second);
```

Comment:
@bmarzocc , is it possible avoiding two loops (lines 481, 491) over hitsAndEnergies and do things in one loop?